src/azure-cli/azure/cli/command_modules/synapse/_params.py

```diff
@@ -196,3 +202,27 @@ def load_arguments(self, _):
         c.argument('code', completer=FilesCompleter(),
                    help='The code of Spark statement. This is either the code contents or use `@<file path>` to load the content from a file')
         c.argument('language', arg_type=get_enum_type(SparkStatementLanguage), validator=validate_statement_language, help='The language of Spark statement.')
+
+    # synapse workspace access-control
+    for scope in ['create', 'list']:
+        with self.argument_context('synapse role assignment ' + scope) as c:
+            c.argument('workspace_name', arg_type=workspace_name_arg_type)
+            c.argument('role', arg_type=role_arg_type)
+            c.argument('assignee', arg_type=assignee_arg_type)
+
+    with self.argument_context('synapse role assignment show') as c:
+        c.argument('workspace_name', arg_type=workspace_name_arg_type)
+        c.argument('role_assignment_id', options_list=['--id'], help='Id of the role that is assigned to the principal.')
+
+    with self.argument_context('synapse role assignment delete') as c:
+        c.argument('workspace_name', arg_type=workspace_name_arg_type)
+        c.argument('role', arg_type=role_arg_type)
+        c.argument('assignee', arg_type=assignee_arg_type)
+        c.argument('ids', nargs='+', help='space-separated role assignment ids')
```

Comment:
Add help message here "You should provide either --ids or --role plus --assignee" so that users can know how to pass the arguments.